uwrt_mars_rover_neopixel/src/uwrt_mars_rover_neopixel_control.cpp

```diff
@@ -0,0 +1,42 @@
+#include "uwrt_mars_rover_neopixel/uwrt_mars_rover_neopixel_control.h"
+
+constexpr uint16_t NEOPIXEL_CAN_ID_OUTGOING = 0x784;
+constexpr uint8_t FRAME_PAYLOAD_LENGTH = 1;
+constexpr uint8_t LOOP_RATE = 1;
+
+Neopixel::Neopixel(int argc, char **argv, char *can_interface)
+    : _arg_count(argc),
+      _arg_list(argv),
+      _can_interface(can_interface),
+      _neopixel_can_msg(NEOPIXEL_CAN_ID_OUTGOING, FRAME_PAYLOAD_LENGTH, can_interface) {
+  ros::init(argc, argv, "neopixel_set_server");
+}
+bool Neopixel::setState(uwrt_mars_rover_msgs::set_state::Request &req, uwrt_mars_rover_msgs::set_state::Response &res) {
+  ROS_INFO("Neopixel callback function triggered.");
+  _state = req.requested_mode.value;
+  _neopixel_can_msg.sendCAN(_state);
+  if (_neopixel_can_msg.waitforAck()) {
+    ROS_INFO("Received acknowledgement from gimbtonomy board.");
+    // git CI insists on using 1u
+    res.success = 1u;
+    return res.success != 0u;
+  }
+  // As currently implemented, this will never run. See waitforAck() for more details
+  ROS_ERROR("Unexpected acknowledgement message received.");
+  res.success = 0u;
```

Comment:
this shold be within the if statement above

Reply:
uhh why? the function returns true if it succeeds and we dont want to raise an error if it succeeds.